File: frontend/src/app/shared/service/admin/reception/ReceptionAdmin.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {BehaviorSubject, Observable} from 'rxjs';

import {environment} from 'src/environments/environment';

import {ReceptionDto} from 'src/app/shared/model/reception/Reception.model';
import {ReceptionCriteria} from 'src/app/shared/criteria/reception/ReceptionCriteria.model';
import {AbstractService} from 'src/app/zynerator/service/AbstractService';


@Injectable({
  providedIn: 'root'
})
export class ReceptionAdminService extends AbstractService<ReceptionDto, ReceptionCriteria> {
     constructor(private http: HttpClient) {
        super();
        this.setHttp(http);
    }

    public constrcutDto(): ReceptionDto {
        return new ReceptionDto();
    }

    public constrcutCriteria(): ReceptionCriteria {
        return new ReceptionCriteria();
    }

    get API() {
        return environment.apiUrlStockyservice + 'admin/reception/';
    }
}

File: backend-ms1/src/main/java/ma/zs/stocky/service/facade/admin/reception/ReceptionAdminService.java
package ma.zs.stocky.service.facade.admin.reception;

import java.util.List;
import ma.zs.stocky.bean.core.reception.Reception;
import ma.zs.stocky.dao.criteria.core.reception.ReceptionCriteria;
import ma.zs.stocky.zynerator.service.IService;



public interface ReceptionAdminService extends  IService<Reception,ReceptionCriteria>  {

    List<Reception> findByCommandeId(Long id);
    int deleteByCommandeId(Long id);
    long countByCommandeRef(String ref);




}
